src/goto-symex/goto_symex_state.cpp

```diff
@@ -187,8 +186,7 @@ bool goto_symex_statet::constant_propagation_reference(const exprt &expr) const
   }
   else if(expr.id()==ID_member)
   {
-    if(expr.operands().size()!=1)
-      throw "member expects one operand";
+    PRECONDITION(expr.operands().size() == 1);
```

Comment:
By definition this falls between multiple categories of the invariant checks (it falls both under `DATA_INVARIANT` because it checks fundamental data structures like an `exprt` in this case, but it also falls under the `PRECONDITION` case, in that it's asserting whether an argument of the function is correct and the rest of the computation can be sound based on that assertion.

But I can certainly change it to the `DATA_INVARIANT` in case it makes things a little bit cleaner.